lib/Atat/Objects.pm

```diff
@@ -0,0 +1,96 @@
+# Domain model.
+#
+# This file describes our domain using an object model.
+#
+# The following terminology may be helpful:
+#     * 'package' defines a class.
+#     * A class 'has' an attribute, which may be another object (delegation).
+#     * Plural attributes imply a 1-many (or many-many) relationship.
+#     * A class 'is' another class (vertical composition).
+#     * A class 'does' a role (horizontal composition).
+
+# User: An ATAT user.
+package User {
+  use Mojo::Base -base;
+  has 'username';    # Username in ATAT
+  has 'piv';         # Data from PIV card.
+  has 'roles';       # Roles within ATAT.
+}
+
+# UserRole: Defines a set of permissions surrounding access to a CSP.
+package UserRole {
+  use Mojo::Base -base;
+  has 'csp_account';    # CSP account (optional)
+  has 'permissions';    # List of permissions (strings) for this role.
+}
+
```

Comment:
@briandds I would suggest that we change the name of this object to "CSPUserRole" so as to not confuse it with "ATATUserRole"

Reply:
Hmm... not sure I agree @hardikdds. We have two types of roles, yes, but we don't have a separate object for the two types, which might be just fine, but if we change this name then we need a new object as well for the other kind of role.

Reply:
@jordangov    Thanks.  And Brian just explained to me how roles in most csps work.